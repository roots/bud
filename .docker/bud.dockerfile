FROM node:lts

LABEL name 'bud'
LABEL version 1

RUN npm install -g verdaccio pm2

COPY ./.docker/bud/motd /etc/motd
COPY ./.docker/bud/bash.bashrc /etc/bash.bashrc
