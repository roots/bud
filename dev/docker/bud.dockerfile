FROM node:16

LABEL name 'bud'
LABEL version 1

RUN npm install netlify-cli npm-cli-login --global

COPY ./dev/docker/bud/motd /etc/motd
COPY ./dev/docker/bud/bash.bashrc /etc/bash.bashrc

COPY ./ /srv/bud

RUN mkdir -p /srv/bud/node_modules \
  && mkdir -p /srv/bud/storage/verdaccio \
  && mkdir -p /srv/bud/storage/yarn \
  && mkdir -p /srv/bud/storage/coverage \
  && mkdir -p /srv/mocks
