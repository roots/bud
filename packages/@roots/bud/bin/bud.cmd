@echo off

node "%~dp0\bud.js" %*
