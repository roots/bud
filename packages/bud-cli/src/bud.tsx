#!/usr/bin/env node
import {yargs} from '@roots/bud-support'
import * as build from './commands/build'
import * as publish from './commands/publish'

/**
 * $bud
 */
const cli = yargs
  .command(build)
  .command(publish)
  .recommendCommands()
  .demandCommand(1, 'Try using one of the above commands.')
  .usage('bud [command] [options]')
  .version()
  .wrap(yargs.terminalWidth())
  .showHelpOnFail(true)
  .epilog('https://github.com/roots/bud').argv

export {cli as bud}
module.exports = {bud: cli}
