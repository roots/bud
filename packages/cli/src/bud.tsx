#!/usr/bin/env node

import {yargs} from '@roots/bud-support'
import * as build from './commands/build'
import * as publish from './commands/publish'

module.exports = yargs
  .command(build)
  .command(publish)
  .recommendCommands()
  .demandCommand(1, 'Try using one of the above commands.')
  .usage('bud [command] [options]')
  .version()
  .wrap(yargs.terminalWidth())
  .showHelpOnFail(true)
  .epilog('https://github.com/roots/bud').argv
