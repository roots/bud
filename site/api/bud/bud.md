---
id: bud.bud
title: bud namespace
sidebar_label: bud namespace
hide_title: true
sidebar: "api"
slug: bud
---

<!-- Do not edit this file. It is automatically generated by API Documenter. -->

## Bud namespace

Signature:

```typescript
export declare namespace Bud
```

## Type Aliases

| Type Alias                      | Description             |
| ------------------------------- | ----------------------- |
| [Options](/api/bud/bud/options) | Bud constructor options |
