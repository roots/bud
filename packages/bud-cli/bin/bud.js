#!/usr/bin/node
require('./../lib/bud')
