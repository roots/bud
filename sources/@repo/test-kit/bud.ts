import * as CONSTANTS from '@repo/constants'
import {Bud, factory as budFactory, makeContext, seed} from '@roots/bud'
import {Config} from '@roots/bud-framework'
import {join} from 'node:path'

export const repoPath = (...path: Array<string>) =>
  join(CONSTANTS.REPO_PATH, ...(path ?? []))

export const mockProject = {
  path: repoPath('tests/util/project'),
}

export const factory = async (options?: Config.Options) => {
  const context = await makeContext(repoPath('tests/util/project'))

  const bud = await budFactory({
    name: 'bud',
    mode: 'production',
    ...(options ?? {}),
    context: {
      ...context,
      ...(options?.context ?? {}),
      args: {
        ...(options?.context?.args ?? {}),
        ci: true,
      },
    },
    seed: {
      ...seed,
      ...(options?.seed ?? {}),
    },
  })

  return bud
}

export {Bud}
