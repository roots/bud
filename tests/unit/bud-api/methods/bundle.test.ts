import {Bud, factory} from '@repo/test-kit/bud'

describe('bud.bundle', () => {
  let bud: Bud

  beforeEach(async () => {
    bud = await factory()
  })

  it('sets from string', async () => {
    bud.bundle('react')

    await bud.api.processQueue()

    expect(
      bud.hooks.filter('build.optimization.splitChunks'),
    ).toMatchSnapshot()
  })

  it('sets from arrayed string', async () => {
    bud.bundle('react', ['react', 'react-dom'])

    await bud.api.processQueue()

    expect(
      bud.hooks.filter('build.optimization.splitChunks'),
    ).toMatchSnapshot()
  })
})
