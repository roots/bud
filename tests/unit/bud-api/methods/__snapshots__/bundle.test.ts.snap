// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bud.bundle sets from arrayed string 1`] = `
Object {
  "automaticNameDelimiter": "/",
  "cacheGroups": Object {
    "react": Object {
      "filename": "js/bundle/react/[name].js",
      "idHint": "react",
      "priority": -10,
      "test": /\\[\\\\/\\]\\(react\\|react-dom\\)\\[\\\\/\\]/,
    },
  },
  "chunks": "all",
  "minSize": 0,
}
`;

exports[`bud.bundle sets from string 1`] = `
Object {
  "automaticNameDelimiter": "/",
  "cacheGroups": Object {
    "react": Object {
      "filename": "js/bundle/react/[name].js",
      "idHint": "react",
      "priority": -10,
      "test": /\\[\\\\/\\]\\(react\\)\\[\\\\/\\]/,
    },
  },
  "chunks": "all",
  "minSize": 0,
}
`;
